caddyconfig/caddyfile/lexer.go

```diff
@@ -138,9 +140,12 @@ func (l *lexer) next() bool {
 			continue
 		}
 
-		if ch == '#' {
+		if ch == '#' && spacePrior {
```

Comment:
Rather than adding a new variable to keep track of spaces, maybe a better approach would be to utilize the existing state, for example, `&& len(val) == 0` instead? Worth looking into, since it'd be simpler and would reuse existing token logic.

Reply:
Ah yeah - `len(val) == 0` works perfectly, I didn't grok how val was used until you mentioned it. 👍